CRM/Event/BAO/Event.php

```diff
@@ -1638,6 +1638,16 @@ public static function displayProfile(&$params, $gid, &$groupTitle, &$values, &$
                   }
                   $skip = TRUE;
                 }
+                // for checkboxes, change array of [key => bool] to array of [idx => key]
+                elseif ($dao->html_type == 'CheckBox') {
+                  $v = [];
+                  foreach ($params[$name] as $key => $val) {
+                    if ($val) {
+                      $v[] = $key;
+                    }
+                  }
+                  $customVal = $v;
```

Comment:
+1 on the concept and the fix.
Nitpick: It looks to me like all of this could be written in 1 line:

    $customVal = array_keys(array_filter($params[$name]));

Reply:
Very possible... i'm not familiar with array_filter :)
I will fix this.